ops/storage-ddls/cassandra-setup.cql

```diff
@@ -309,6 +309,21 @@ AS SELECT periodtype, conjunctiontopic1, conjunctiontopic2, conjunctiontopic3, t
 PRIMARY KEY ((periodtype, conjunctiontopic1, conjunctiontopic2, conjunctiontopic3, tilez, pipelinekey, tileid), perioddate, mentioncount, externalsourceid)
 WITH CLUSTERING ORDER BY (perioddate DESC, mentioncount DESC);
 
+CREATE MATERIALIZED VIEW eventsbypipeline
+AS SELECT eventid, pipelinekey, eventtime,conjunctiontopic3,conjunctiontopic2,tilez,tileid,placeid,conjunctiontopic1
+   FROM eventplaces
+   WHERE eventid IS NOT NULL
+     AND pipelinekey IS NOT NULL
+     AND eventtime IS NOT NULL
+     AND conjunctiontopic3 IS NOT NULL
+     AND conjunctiontopic2 IS NOT NULL
+     AND tilez IS NOT NULL
+     AND tileid IS NOT NULL
+     AND placeid IS NOT NULL
+     AND conjunctiontopic1 IS NOT NULL
+PRIMARY KEY ((pipelinekey), eventtime, eventid, conjunctiontopic1, conjunctiontopic2, conjunctiontopic3, tilez, tileid, placeid)
```

Comment:
That MV would result into large data partitions, and slow queries.
 
Please set the PK to the following `PRIMARY KEY ((pipelinekey, conjunctiontopic1, conjunctiontopic2, conjunctiontopic3, tilez), eventtime, eventid, tileid, placeid)`

when you need to query for facts you can query by a designated tilez = 15. So for example, 
`Select * from eventplaces where piplelinekey in ('Twitter', 'Facebook') and tilez = 15 and conjunctiontopic1 IN('{INCLUDE ALL TOPIC TERMS}') and conjunctiontopic2 = '' and conjunctiontopic3 = '' and eventtime > '12/21/2017' and eventtime < 12/31/2017`. 

You may even want to consider setting `conjunctiontopic2 = '' and conjunctiontopic3 = ''` within the MV where clause.

Reply:
We could do that, but in that case, do we really need this view? If we restrict events by time/topics/etc. how would this component be different from the ActivityFeed component when the map is maximally zoomed out?

Reply:
Thinking about this a bit more, maybe a nice way to partition this data would be by date (day/hour-level). In that way, we can implement the infinite scroll very nicely by just requesting progressively farther back dates: every infinite scroll request then maps to a partition.

Reply:
Pinging @erikschlegel: what's the word on the facts view? Do we really want to implement a separate page for the Facts view if it'll simply surface the same information as the NewsFeed when the map is maximally zoomed out?

I'm happy to implement this, just worried that this will add extra maintenance burden for the same information for which we already have a view.